experimental/bits/simd_x86.h

```diff
@@ -4235,21 +4235,21 @@ struct _MaskImplX86 : _MaskImplX86Mixin, _MaskImplBuiltin<_Abi>
 	    __m128i __a = {};
 	    __builtin_memcpy(&__a, __mem, 16);
 	    const auto __b = _mm512_cvtepi8_epi32(__a);
-	    __builtin_memcpy(&__a, __mem + 16, 16);
+	    __builtin_memcpy(&__a, static_cast<const char*>(__mem) + 16, 16);
 	    const auto __c = _mm512_cvtepi8_epi32(__a);
 	    if constexpr (size<_Tp> <= 48)
 	      {
-		__builtin_memcpy(&__a, __mem + 32, size<_Tp> - 32);
+		__builtin_memcpy(&__a, static_cast<const char*>(__mem) + 32, size<_Tp> - 32);
 		const auto __d = _mm512_cvtepi8_epi32(__a);
 		return _mm512_test_epi32_mask(__b, __b)
 		       | (_mm512_test_epi32_mask(__c, __c) << 16)
 		       | (_ULLong(_mm512_test_epi32_mask(__d, __d)) << 32);
 	      }
 	    else
 	      {
-		__builtin_memcpy(&__a, __mem + 16, 32);
+		__builtin_memcpy(&__a, static_cast<const char*>(__mem) + 16, 32);
```

Comment:
my compiler flags this operation to always overflow, since __m128i __a is 16 bytes thing

Reply:
This looks broken, yes. I'll take a look.

Reply:
Yes, this needs to be 16. Seems I have no test coverage for this case :wink:

Reply:
There is already some value from clang then :)